diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/FunctonASTNodeUtils.kt

```diff
@@ -1,11 +1,36 @@
 package org.cqfn.diktat.ruleset.utils
 
 import com.pinterest.ktlint.core.ast.ElementType
+import com.pinterest.ktlint.core.ast.ElementType.ANNOTATION_ENTRY
+import com.pinterest.ktlint.core.ast.ElementType.IDENTIFIER
+import com.pinterest.ktlint.core.ast.ElementType.MODIFIER_LIST
+import org.cqfn.diktat.ruleset.rules.PackageNaming.Companion.PACKAGE_PATH_ANCHOR
 import org.jetbrains.kotlin.com.intellij.lang.ASTNode
 import org.jetbrains.kotlin.kdoc.parser.KDocKnownTag
 import org.jetbrains.kotlin.psi.KtParameter
 import org.jetbrains.kotlin.psi.KtParameterList
 
+/**
+ * Checks whether function from this [ElementType.FUN] node is from test class based on annotations and file location
+ */
+fun ASTNode.isTestFun(filePathParts: List<String>): Boolean {
+    checkNode(this)
+    val hasTestAnnotation = findChildByType(MODIFIER_LIST)?.getAllChildrenWithType(ANNOTATION_ENTRY)
+            ?.any { it.findLeafWithSpecificType(IDENTIFIER)?.text == "Test" }
+            ?: false
+    val isLocatedInTest =
+            filePathParts
+                    .takeIf { it.contains(PACKAGE_PATH_ANCHOR) }
+                    ?.run { subList(lastIndexOf(PACKAGE_PATH_ANCHOR), size) }
+                    ?.let {
+                        // e.g. src/test/ClassTest.kt
+                        it.contains("test") || it.last().substringBeforeLast('.').endsWith("Test")
```

Comment:
you know what I am thinking? sometimes in test directories there could be utils files (frameworks/e.t.c)
May be we will add contains("test") && endsWith("Test")

Reply:
Makes sense.

Reply:
Actually, why should we check directory at all then? If not all functions located under `test` are ignored, and taking into account that users can have different test directories (e.g. `jvmTest` in kotlin multiplatform), can we just check that file ends with `Test`?